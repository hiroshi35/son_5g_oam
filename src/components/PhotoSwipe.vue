<template>
  <div>
    <div class="sc-gallery" ref="thumbnails">
      <figure v-for="(item, index) in items" @click.prevent="openGallery(index, $event)">
        <a :href="item.src">
          <img :src="item.src"/>
        </a>
        <figcaption>{{item.title}}</figcaption>
      </figure>
    </div>
    <div class="pswp" tabindex="-1" role="dialog" aria-hidden="true" ref="gallery">
      <div class="pswp__bg"></div>
      <div class="pswp__scroll-wrap">
        <div class="pswp__container">
          <div class="pswp__item"></div>
          <div class="pswp__item"></div>
          <div class="pswp__item"></div>
        </div>
        <div class="pswp__ui pswp__ui--hidden">
          <div class="pswp__top-bar">
            <div class="pswp__counter"></div>
            <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
            <button class="pswp__button pswp__button--share" title="Share"></button>
            <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
            <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
            <div class="pswp__preloader">
              <div class="pswp__preloader__icn">
                <div class="pswp__preloader__cut">
                  <div class="pswp__preloader__donut"></div>
                </div>
              </div>
            </div>
          </div>
          <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
            <div class="pswp__share-tooltip"></div>
          </div>
          <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
          </button>
          <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
          </button>
          <div class="pswp__caption">
            <div class="pswp__caption__center"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
<script>
  import PhotoSwipe from 'photoswipe'
  import PhotoSwipeUIDefault from 'photoswipe/dist/photoswipe-ui-default'

  export default {
    props: {
      items: {
        type: Array,
        required: true
      },
      options: {
        type: Object,
        default () {
          return {}
        }
      }
    },
    data () {
      return {
        configuration: this.options
      }
    },
    methods: {
      openGallery (index) {
        let vm = this
        this.configuration.index = index
        this.configuration.getThumbBoundsFn = function (index) {
          let rect = $(vm.$refs.thumbnails).children()[index].getBoundingClientRect()
          let pageYScroll = window.pageYOffset || document.documentElement.scrollTop
          return {x: rect.left, y: rect.top + pageYScroll, w: rect.width}
        }
        // Initializes and opens PhotoSwipe
        let gallery = new PhotoSwipe(this.$refs.gallery, PhotoSwipeUIDefault, this.items, this.configuration)
        gallery.init()
      }
    }
  }
</script>

<style lang="scss">
  @import "~photoswipe/dist/photoswipe";
  @import "~photoswipe/dist/default-skin/default-skin";

  .sc-gallery {
    width: 100%;
    float: left;
  }

  .sc-gallery img {
    width: 100%;
    height: auto;
  }

  .sc-gallery figure {
    display: block;
    float: left;
    margin: 0 5px 5px 0;
    width: 150px;
  }

  .sc-gallery figcaption {
    display: none;
  }
  .pswp__caption__center{
    text-align: center;
  }
</style>
