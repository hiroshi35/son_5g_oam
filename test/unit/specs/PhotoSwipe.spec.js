import { shallow } from 'vue-test-utils'
import PhotoSwipe from '@/components/PhotoSwipe.vue'
import {ps, init} from 'photoswipe'

describe('PhotoSwipe.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(PhotoSwipe, {
      propsData: {
        items: [
          {
            src: '/static/imgs/carousel/1.jpg',
            w: 5184,
            h: 2400,
            title: 'First Image',
            author: 'John Doe'
          }
        ]
      }
    })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('opens gallery', () => {
    wrapper.find('figure').trigger('click')
    expect(ps).toBeCalled()
    expect(init).toBeCalled()
  })
})
