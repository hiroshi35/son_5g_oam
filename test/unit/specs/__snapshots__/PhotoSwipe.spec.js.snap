// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoSwipe.vue has the expected html structure 1`] = `
<div>
  <div
    class="sc-gallery"
  >
    <figure>
      <a
        href="/static/imgs/carousel/1.jpg"
      >
        <img
          src="/static/imgs/carousel/1.jpg"
        />
      </a>
       
      <figcaption>
        First Image
      </figcaption>
    </figure>
  </div>
   
  <div
    aria-hidden="true"
    class="pswp"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="pswp__bg"
    />
     
    <div
      class="pswp__scroll-wrap"
    >
      <div
        class="pswp__container"
      >
        <div
          class="pswp__item"
        />
         
        <div
          class="pswp__item"
        />
         
        <div
          class="pswp__item"
        />
      </div>
       
      <div
        class="pswp__ui pswp__ui--hidden"
      >
        <div
          class="pswp__top-bar"
        >
          <div
            class="pswp__counter"
          />
           
          <button
            class="pswp__button pswp__button--close"
            title="Close (Esc)"
          />
           
          <button
            class="pswp__button pswp__button--share"
            title="Share"
          />
           
          <button
            class="pswp__button pswp__button--fs"
            title="Toggle fullscreen"
          />
           
          <button
            class="pswp__button pswp__button--zoom"
            title="Zoom in/out"
          />
           
          <div
            class="pswp__preloader"
          >
            <div
              class="pswp__preloader__icn"
            >
              <div
                class="pswp__preloader__cut"
              >
                <div
                  class="pswp__preloader__donut"
                />
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap"
        >
          <div
            class="pswp__share-tooltip"
          />
        </div>
         
        <button
          class="pswp__button pswp__button--arrow--left"
          title="Previous (arrow left)"
        />
         
        <button
          class="pswp__button pswp__button--arrow--right"
          title="Next (arrow right)"
        />
         
        <div
          class="pswp__caption"
        >
          <div
            class="pswp__caption__center"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
